### PR TITLE
(feat) add sort capability to SkawaDataTable

### PR DESCRIPTION
**Pull request by szepeshazi**

### CODE REVIEW

**Comment by valakis** [comment]
update CHANGELOG and README

File: lib/src/components/data_table/data_table.dart [hunk 1]
```diff
@@ -74,10 +79,16 @@ class SkawaDataTableComponent implements OnDestroy {
   @Output('highlight')
   Stream<RowData> get onHighlight => _highlightController.stream;
 
+  @Output()
```
Comment (valakis) [inline lib/src/components/data_table/data_table.dart]: the output should be `'sort'`

File: test/data_table_test.dart [hunk 2]
```diff
@@ -394,6 +468,22 @@ class TableCellPO {
   PageLoaderElement materialCheckbox;
 }
 
+class TableHeaderCellPO {
+  @inject
+  PageLoader loader;
```
Comment (valakis) [inline test/data_table_test.dart]: this loader is unused

Comment (valakis) [reply test/data_table_test.dart]: I know its just copied from the other class, and its my fault to left there this unnecessary filed

File: lib/src/components/data_table/sort.dart [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    }
+    if (directions.isEmpty || directions.any((s) => directionMap[s] == null)) {
+      throw new ArgumentError(
+          'SkawaDataTableSortDirective accepts only "asc" and/or "desc" as sort directions. Use comma separated values for both directions.');
+    }
+    column.sort = new SortModel(directions.map((s) => directionMap[s]).toList(growable: false));
+  }
+
+  @Input()
+  set defaultSort(String sort) {
```
Comment (valakis) [inline lib/src/components/data_table/sort.dart]: there are no way to preserve multiply column to have `defaultSort`

File: lib/src/components/data_table/sort.dart [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+  bool get isAscending => activeSort == SortDirection.asc;
+
+  bool get isDescending => activeSort == SortDirection.desc;
+
+  bool get isSorted => activeSort != null;
+}
+
+@Directive(selector: '[sort]')
+class SkawaDataTableSortDirective {
```
Comment (valakis) [inline lib/src/components/data_table/sort.dart]: Wouldn't be easier just define it in the column definition and additionally add the sort fuction to the column?

File: lib/src/components/data_table/data_table.dart (hunk 1 above)
Comment (szepeshazi) [reply lib/src/components/data_table/data_table.dart]: Funny, this is what I did originally, but I encountered with a strange naming collision for `sort`, with the output, the directive input and the data table SortModel variable name all being identical. But you're right, this should be `sort`, so I renamed the directive input to `sortable` and the `SortModel` variable to `sortModel`. Fixed.

File: test/data_table_test.dart (hunk 2 above)
Comment (szepeshazi) [reply test/data_table_test.dart]: Fixed, but only for this class. Maybe need to do a future pass and check all unused properties.

File: lib/src/components/data_table/sort.dart (hunk 3 above)
Comment (szepeshazi) [reply lib/src/components/data_table/sort.dart]: `defaultSort` is to mark a column for initial sorting in the table. There can be only one sorted column at a given time in the table, so this is intentional. However the naming may be confusing a bit, so I renamed this Input to `initialSort`. For preferred sort direction, you can pass `sortable="asc, desc"` or `sortable="desc, asc"` to the directive.

File: lib/src/components/data_table/sort.dart (hunk 4 above)
Comment (szepeshazi) [reply lib/src/components/data_table/sort.dart]: Maybe so. My intention was to implement sort as a (mostly) reusable add-on on top of the data table, and decouple it from the table as much as possible. Consider, for example, if we later on want to sort grid items (a bit far fetched example, but you get the idea).

If you still think it's better to move everything sort related into the column, I can do it.

File: lib/src/components/data_table/sort.dart (hunk 3 above)
Comment (valakis) [reply lib/src/components/data_table/sort.dart]: I wanted to write prevent but i writed preserve

Comment (szepeshazi) [reply lib/src/components/data_table/sort.dart]: Fixed

**Pull request merged**